src/Engine/Component/AnimationController.cpp

```diff
@@ -0,0 +1,92 @@
+#include "AnimationController.hpp"
+#include "../Animation/Skeleton.hpp"
+#include "../Animation/AnimationClip.hpp"
+#include <Utility/Log.hpp>
+#include "glm/gtc/quaternion.hpp"
+
+using namespace Component;
+
+AnimationController::AnimationController() {
+    for (unsigned int i = 0; i < 20; ++i) {
+        bones.push_back(glm::mat4(1.0f));
+    }
+}
+
+Json::Value AnimationController::Save() const {
+    Json::Value component;
+
+    if (controller != nullptr)
+        component["animationController"] = controller->path + controller->name;
+
+    if (skeleton != nullptr)
+        component["skeleton"] = skeleton->path + skeleton->name;
+
+    return component;
+}
+
+void Component::AnimationController::UpdateAnimation(float deltaTime) {
+    if (skeleton == nullptr || controller == nullptr)
+        return;
+
+    if (bones.size() != skeleton->skeletonBones.size()) {
+        bones.clear();
+        bones.shrink_to_fit();
```

Comment:
See earlier comments.

Reply:
This is necessary as I was getting the wrong size of the array by just clearing and resizing it.